### PR TITLE
fix: multiple minor issue fixes

### PR DESCRIPTION
**Pull request by rish2320**
# Description

Fixing multiple minor issues 

Fixes #[2570](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=2570)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
1: I tested it by going through the other functions or components where the change might be reflected and seeing what would be the optimal way to solve the error.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const getManifestResource = (
 }
 
 export const getDesiredManifestResource = (appDetails: AppDetails, podName: string, nodeType: string) => {
-    const selectedResource = appDetails.resourceTree.nodes.filter(
+    const selectedResource = appDetails.resourceTree?.nodes.filter(
```
Comment (milindk8) [inline src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts:33]: the value will be undefined, is there any handling further in the code

File: src/components/ciPipeline/validationRules.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import {
 export class ValidationRules {
     name = (value: string): { message: string | null; isValid: boolean } => {
         const regExp = new RegExp(PATTERNS.APP_NAME)
-        if (value.length === 0) return { isValid: false, message: REQUIRED_FIELD_MSG }
+        if (!value || value.length === 0) return { isValid: false, message: REQUIRED_FIELD_MSG }
```
[the remark this answers is not in the document]
Comment (rish2320) [reply src/components/ciPipeline/validationRules.ts:13]: done

File: src/components/app/details/appConfig/AppConfig.tsx [hunk 3]
```diff
@@ -361,7 +361,7 @@ export default function AppConfig({ appName, isJobView }: AppConfigProps) {
                 .slice()
                 .reverse()
                 .find((stage) => stage.status)
-            _lastConfiguredStage = lastConfiguredStage.stageName
+            _lastConfiguredStage = lastConfiguredStage.stageName ? lastConfiguredStage.stageName : ''
```
[the remark this answers is not in the document]
Comment (rish2320) [reply src/components/app/details/appConfig/AppConfig.tsx:364]: done

File: src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts (hunk 1 above)
Comment (rish2320) [reply src/components/v2/appDetails/k8Resource/nodeDetail/nodeDetail.api.ts:33]: Yes

File: src/components/app/details/triggerView/TriggerView.tsx [hunk 4]
```diff
@@ -996,7 +996,7 @@ class TriggerView extends Component<TriggerViewProps, TriggerViewState> {
                                 stageType={DeploymentNodeType[this.state.nodeType]}
                                 material={material}
                                 materialType={this.state.materialType}
-                                envName={node.environmentName}
+                                envName={node?.environmentName}
```
Comment (milindk8) [inline src/components/app/details/triggerView/TriggerView.tsx:999]: have you tested for envName as undefined

File: src/services/service.ts [hunk 5]
```diff
@@ -382,7 +382,7 @@ export function getAppChartRef(appId: number): Promise<ResponseType> {
             result: { chartRefs, latestAppChartRef },
         } = response
         let selectedChartId = latestAppChartRef
-        let chart = chartRefs.find((chart) => selectedChartId === chart.id)
+        let chart = chartRefs?.find((chart) => selectedChartId === chart.id)
```
Comment (milindk8) [inline src/services/service.ts:385]: if the chart will be undefined, investigate whether an empty value affects further code

File: src/components/app/details/triggerView/TriggerView.tsx (hunk 4 above)
Comment (rish2320) [reply src/components/app/details/triggerView/TriggerView.tsx:999]: Checked the envName to be undefined doesn't cause any error.

File: src/components/ClusterNodes/NodeList.tsx [hunk 6]
```diff
@@ -356,13 +356,15 @@ export default function NodeList({ imageList, isSuperAdmin, namespaceList }: Clu
     }
 
     const alphabeticalComparatorMethod = (a, b) => {
+        let firstValue = a[sortByColumn.sortingFieldName] || ""
```
[the remark this answers is not in the document]
Comment (rish2320) [reply src/components/ClusterNodes/NodeList.tsx]: done

**Review by milindk8 — APPROVED** [review]

File: src/components/ResourceBrowser/ResourceList/ResourceList.tsx [hunk 7]
```diff
@@ -369,7 +369,7 @@ export default function ResourceList() {
             type: 'start',
             payload: {
                 searchText: _searchText,
-                list: _resourceList.data,
+                list: _resourceList.data || [],
```
Comment (vivek-devtron) [inline src/components/ResourceBrowser/ResourceList/ResourceList.tsx:372]: we can add check of _resourceList on line no 367 may be

File: src/components/app/details/appConfig/AppConfig.tsx [hunk 8]
```diff
@@ -361,7 +361,7 @@ export default function AppConfig({ appName, isJobView }: AppConfigProps) {
                 .slice()
                 .reverse()
                 .find((stage) => stage.status)
-            _lastConfiguredStage = lastConfiguredStage.stageName
+            _lastConfiguredStage = lastConfiguredStage?.stageName || ''
```
Comment (vivek-devtron) [inline src/components/app/details/appConfig/AppConfig.tsx:364]: we can replicate the line no 349 check here. i.e. if(!lastConfiguredStage){

File: src/components/ResourceBrowser/ResourceList/ResourceList.tsx (hunk 7 above)
Comment (rish2320) [reply src/components/ResourceBrowser/ResourceList/ResourceList.tsx:372]: done

File: src/components/app/details/appConfig/AppConfig.tsx (hunk 8 above)
Comment (rish2320) [reply src/components/app/details/appConfig/AppConfig.tsx:364]: done

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**